lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerendpoint.py

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Hai Cao, <t-haicao@microsoft.com>, Yunge Zhu <yungez@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_trafficmanagerendpoint
+version_added: "2.7"
+short_description: Manage Azure Traffic Manager endpoint.
+description:
+    - Create, update and delete Azure Traffic Manager endpoint.
+
+options:
+    resource_group:
+        description:
+            - Name of a resource group where the Traffic Manager endpoint exists or will be created.
+        type: str
+        required: true
+    name:
+        description:
+            - The name of the endpoint.
+        type: str
+        required: true
+    profile_name:
+        description: Name of Traffic Manager profile where this endpoints attaches to.
+        type: str
+        required: true
+    type:
+        description:
+            - The type of the endpoint.
+        required: true
+        choices:
+            - azureEndpoints
+            - externalEndpoints
+            - nestedEndpoints
+    target_resource_id:
+        description:
+            - The Azure Resource URI of the of the endpoint.
+            - Not applicable to endpoints of type 'ExternalEndpoints'.
+        type: str
+    target:
+        description:
+            - The fully-qualified DNS name of the endpoint.
+        type: str
+    enable:
+        description:
+            - The status of the endpoint.
+        type: bool
+        default: true
+    weight:
+        description:
+            - The weight of this endpoint when using the 'Weighted' traffic routing method.
+            - Possible values are from 1 to 1000.
+        type: int
+    priority:
+        description:
+            - The priority of this endpoint when using the 'Priority' traffic routing method.
+            - Possible values are from 1 to 1000, lower values represent higher priority.
+            - This is an optional parameter. If specified, it must be specified on all endpoints.
+            - No two endpoints can share the same priority value.
+        type: int
+    location:
+        description:
+            - Specifies the location of the external or nested endpoints when using the 'Performance' traffic routing method.
+        type: str
+    min_child_endpoints:
+        description:
+            - The minimum number of endpoints that must be available in the child profile in order for the parent profile to be considered available.
+            - Only applicable to endpoint of type 'NestedEndpoints'.
```

Comment:
`Only application to an endpoint of I(type) C(nested_endpoints).

Reply:
fixed